portable/GCC/ARM_CRx_MPU/mpu_wrappers_v2_asm.S

```diff
@@ -0,0 +1,1333 @@
+/*
+ * FreeRTOS Kernel <DEVELOPMENT BRANCH>
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+
+/*---------------------------------------------------------------------------*/
+
+    .arm
+    .syntax unified
+    .section freertos_system_calls
+#define FREERTOS_ASSEMBLY
+    #include "FreeRTOSConfig.h"
+    #include "portmacro_asm.h"
+    #include "mpu_syscall_numbers.h"
+#undef FREERTOS_ASSEMBLY
+
+/* ------------------ Start of Port Specific System Calls ------------------ */
```

Comment:
It looks like the majority of the functionality provided by this file (and associated header) are to support the mpu v2 wrappers. However, there are a few functions in this file which seem like they would be better placed in portASM.s such as vPortYied, vPortSystemCallExit, xPortIsPrivileged, ulPortCountLeadingZeros for consistency.

Reply:
The portASM.S is marked  as PRIVILEGED_FUNCTIONS to ensure that they placed in the correct section of the linker script

Reply:
Due to this any system calls need to be placed into mpu_wrappers_v2.S to ensure they are accessible to unprivileged tasks